src/clj/rems/application/eraser.clj

```diff
@@ -0,0 +1,63 @@
+(ns rems.application.eraser
+  (:require [mount.core :as mount]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [rems.scheduler :as scheduler]
+            [rems.db.applications :as applications]
+            [rems.config :refer [env]])
+  (:import [org.joda.time Period]))
+
+(defn- is-expired?
+  [application]
+  (let [last-activity (:application/last-activity application)
+        state (:application/state application)
+        expiration-threshold (some-> env :application-expiration state Period/parse)]
+    (when expiration-threshold
+      (time/before? last-activity (time/minus (time/now) expiration-threshold)))))
+
+(deftest test-is-expired
+  (with-redefs [env {:application-expiration {:application.state/draft "P90D"
+                                              :application.state/closed "P7D"}}]
+    (testing "should identify expired application by state"
+      (let [over-90d-ago (time/minus (time/now) (time/days 90) (time/seconds 1))
+            over-7d-ago (time/minus (time/now) (time/days 7) (time/seconds 1))
+            over-1d-ago (time/minus (time/now) (time/days 1) (time/seconds 1))]
+        (is (true? (is-expired? {:application/state :application.state/draft
+                                 :application/last-activity over-90d-ago})))
+        (is (false? (is-expired? {:application/state :application.state/draft
+                                  :application/last-activity over-7d-ago})))
+        (is (true? (is-expired? {:application/state :application.state/closed
+                                 :application/last-activity over-7d-ago})))
+        (is (false? (is-expired? {:application/state :application.state/closed
+                                  :application/last-activity over-1d-ago})))
+        (is (nil? (is-expired? {:application/state :application.state/rejected
+                                :application/last-activity over-90d-ago})))))))
```

Comment:
i think returning `nil` from the `is-expired?` function might be enough to indicate that the application state was not matched? it still coerces to `false` when evaluated as boolean, so net effect is the same.

Reply:
We can coerce the result in `is-expired?` to be a boolean if we expect that function to be used in many places. Otherwise I'm happy with the `truthy` interpretation and returning `nil` as `false`.

Now these tests make it appear that there should be a difference between `nil` and `false` and I think perhaps we want to avoid it?

Reply:
yeah, you are probably right that we might want to avoid assigning this type difference. i kind of thought that semantic difference would justify separating `false` (= application state matches config but is not expired) and `nil` (application state does not match config).

i'll change the code according to the other comment so it should always evaluate to a boolean.